net/protocol/replicator.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2025 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package protocol
+
+import (
+	"context"
+	"errors"
+
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+
+	"github.com/sourcenetwork/defradb/event"
+	"github.com/sourcenetwork/defradb/net/message"
+)
+
+// PushLogRequest is the struct used to send a resource update to a peer node
+type PushLogRequest struct {
+	message.MetaData
+	DocID        string
+	CID          []byte
+	CollectionID string
+	Creator      string
+	Block        []byte
+}
+
+// PushLogReply is the expected response struct that should be received after
+// an pushlog request.
+type PushLogReply struct {
+	message.MetaData
+}
+
+const (
+	replicatorProtocolRequest  = "/defradb/rep_req/0.0.1"
+	replicatorProtocolResponse = "/defradb/rep_resp/0.0.1"
```

Comment:
Suggestion: similar to Andy's Suggestion above.

Reply:
Done